### PR TITLE
Standardized tide gauge objects

### PR DESCRIPTION
**Pull request by soutobias**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been reviewed and added / updated if needed (for bug fixes / features)
    - Some docs updates need to be made in the `COAsT-site` repo, in a separate PR. See [contributing to documentation ](https://british-oceanographic-data-centre.github.io/COAsT/docs/contributing-docs/) for details.
- [ ] Build (`./build.sh`) was run locally and no errors reported. NB not sure about this requirement: GitActions test this
- [X] Lint (`pylint .`) has passed locally and any fixes were made for failures. NB not sure about this requirement: GitActions test this with `black`

## Pull request type

Please check the type of change your PR introduces:
- [X] Bugfix
- [X] Feature
- [X] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Some lint problems, deprec  functions and not-standard Tidegauge objects

## What is the new behavior?

I made the following changes in the code:

1. Standardized tide gauge objects (gesla, HLW, BODC) by ensuring consistent dimensions, coordinates, variables, and attributes. Renamed id_station to site_name and added the ssh variable. This is the standard:

image

2. Optimized methods in HLW, removing unnecessary loops in file opening.

3. There's an issue converting datetime from Europe/London to UTC because certain datetimes, like "25/10/2020 01:18," can be ambiguous (BST or not). While you previously encountered this with pytz without noticing, using pandas now highlights this ambiguity. I employed the pytz convention treating ambiguous datetimes as BST, but I'm uncertain if this is the optimal solution.

4. Updated tests, ensuring all pass successfully.

5. Revised notebooks and added an example in tidegauge validation on opening BODC files.

6. Flagged the deprecated method `read_gesla_v3` and the class `Process_data` with a deprecation decorator. Updated the class name to camel case ("ProcessData").

7. Corrected linting issues in the code.

### Checklist
- [ ] Code compiles correctly
- [ ] Created tests which fail without the change (if possible)
- [X] All tests passing (`pip install . && pytest unit_testing/unit_test.py -s`)
- [ ] Extended the README / documentation, if necessary
- [ ] Added myself / the copyright holder to the AUTHORS file

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

**Comment by soutobias** [comment]
@jpolton , could you please take a look if everything is okay? Please see the comments that I included in the new behavior section of the first comment.

File: coast/__init__.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 from .diagnostics.tidegauge_analysis import TidegaugeAnalysis
 from .data.config_parser import ConfigParser
 from ._utils.xesmf_convert import xesmf_convert
-from ._utils.process_data import Process_data
+from ._utils.process_data import Process_data, ProcessData
```
Comment (jpolton) [inline coast/__init__.py:26]: `Process_data` has been renamed

Comment (jpolton) [reply coast/__init__.py:26]: Ah. Process_date pipes to ProcessData with a deprecation warning. Clever!

File: coast/data/tidegauge.py [hunk 2]
```diff
@@ -592,13 +567,12 @@ def show(self, timezone: str = None):
...
         else:  # display timezone aware times
-            for ssh, idx in enumerate(self.dataset.ssh):
-                # debug('time:', self.dataset.time[i].values,
+            for idx, ssh in enumerate(self.dataset.ssh):
                 debug(
                     "time (" + timezone + "):",
                     general_utils.day_of_week(self.dataset.time[idx].values),
                     np.datetime_as_string(self.dataset.time[idx], unit="m", timezone=pytz.timezone(timezone)),
-                    "height:",
+                    "heighdxt:",
```
Comment (jpolton) [inline coast/data/tidegauge.py:575]: typo: keep "height"

File: coast/data/tidegauge.py [hunk 3]
```diff
@@ -592,13 +567,12 @@ def show(self, timezone: str = None):
                     "m",
                 )
         else:  # display timezone aware times
-            for ssh, idx in enumerate(self.dataset.ssh):
```
Comment (jpolton) [inline coast/data/tidegauge.py:595]: Golly! Shame on me!

**Pull request merged**